dask/array/utils.py

```diff
@@ -246,6 +258,8 @@ def _get_dt_meta_computed(x, check_shape=True, check_graph=True):
             assert x.dtype == x_original.dtype
         if check_shape:
             assert_eq_shape(x_original.shape, x.shape, check_nan=False)
+        if check_chunks:
+            _check_chunks(x_original)
```

Comment:
Would it be possible to compute only once?  I'm concerned about doubling the compute time of every dask array test

We might call persist at the beginning of the assert_eq function, and then call `.compute()` on that result just afterwards so that we have the original, persisted, and computed versions.

Or, more broadly, how does this change affect the runtime of `py.test dask/array` ?

Reply:
Good question - I wasn't sure which way was better. It goes from 51s to 69s (with the synchronous scheduler for persist). Not double, but not trivial either. I'll try it the other way and see what the difference is.

Is there much risk that replacing `x.compute()` with `x.persist().compute()` will affect test coverage? I'm not really up to speed on the postprocessing protocols used in each case.

Reply:
I've changed it to your suggestion, and now it only adds about 2-3s. It also made the zarr failure go away; my guess is that since the test was to do with writing to a zarr array, it wasn't idempotent and so computing twice failed.